### DISCUSSION TITLE
Using IDEs for developing EDC

### DISCUSSION THREAD

**Discussion by rohrbachd** [eclipse-edc/Connector#3671] [Q&A]
Does anyone has any experience with different IDEs for developing EDC Connector? Are there any documentation available on how to best setup a development environment for the project? 

I am having problems with **VSCode**, which turns out to be very slow for the initial builds and setup (several hours in background tasks) and later startups (several minutes until java is ready and unit tests are recognized). VSCode will eat up a lot of resources of my computer.

**Eclipse** seems to work OK but is not really responsive especially for debugging.

The best so far is **IntelliJ IDEA** (community edition). However, it will not work out of the box. It will fail when trying to build the project with gradle. The reason is a case sensitive name issue in `settings.gradle.kts`   

The file has the entry at the top:
`rootProject.name = "connector"`

which should be 
`rootProject.name = "Connector"`

It will fail with the message
 java.lang.IllegalStateException: Module entity with name: Connector should be available

**Comment by ndr-brt** [comment]
IntelliJ IDEA here, never had any particular issue.

**Reply by rohrbachd** [reply]
nice, what platform do you use? I had to change `rootProject.name = "connector" ` to `rootProject.name = "Connector"` in settings.gradle (line 18) or otherwise it wouldn't compile using gradle from within the IDE (IntelliJ IDEA). 

I would get:
`java.lang.IllegalStateException: Module entity with name: Connector.extensions.control-plane.store.sql.contract-definition-store-sql should be available`

**Reply by jimmarino** [reply]
Neither have I (Intellij, macOS). Sounds like a Windows-related bug in IntelliJ.

**Reply by rohrbachd** [reply]
Thanks for the replies it looks like others had similar issues (https://youtrack.jetbrains.com/issue/IDEA-317606/Changing-only-the-case-of-the-Gradle-root-project-name-causes-exception-while-importing-project-java.lang.IllegalStateException). It looks like if you checkout the project in a folder called `Connector` you will get the error.` rootProject.name = "connector"` must match the project folder name. this is confusing because the repository name is also upper case.

**Reply by ndr-brt** [reply]
my local directory is `Connector`, no issues
